tools/driver/p4c_src/main.py

```diff
@@ -136,69 +154,134 @@ def main():
+    ### DRYified “env_indicates_developer_build”
+    env_indicates_developer_build = os.environ['P4C_BUILD_TYPE'] == "DEVELOPER"
+    if env_indicates_developer_build:
         add_developer_options(parser)
 
-    parser.add_argument("source_file", nargs='?', help="Files to compile", default=None)
+    parser.add_argument("P4_source_files", nargs='*',
+                        help="P4 source files to compile.", default=None)
 
     # load supported configuration.
     # We load these before we parse options, so that backends can register
     # proprietary options
     cfg_files = glob.glob("{}/*.cfg".format(os.environ['P4C_CFG_PATH']))
-    cfg = config.Config(config_prefix = "p4c")
+    cfg = config.Config(config_prefix="p4c")
     for cf in cfg_files:
         cfg.load_from_config(cf, parser)
+    ### ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    ###
+    ### Abe`s note: I hope my {“source_file” → “P4_source_files”} change
+    ###   (to the “parser” object) at this point in time can`t cause any
+    ###   ill effects in the callee “config.Config.load_from_config”.
+    ###
+    ###   WIP To Do: check with somebody more experienced with this codebase
+    ###              before even _thinking_ about merging
+    ###              this changeset into “main”.
 
     # parse the arguments
     opts = parser.parse_args()
 
+
+
     user_defined_version = os.environ.get('P4C_DEFAULT_VERSION')
-    if user_defined_version != None:
+    ### Note: the next line of real code _was_τhe clear and easy-to-understand
+    ###       “if user_defined_version != None:”, but “pycodestyle”
+    ###       complained about that idiom so I rewrote it to something
+    ###       semantically-equivalent but with different syntax.
+    if not (user_defined_version is None):
         opts.language = user_defined_version
-    # accept multiple ways of specifying which language, and ensure that it is a consistent
-    # string from now on.
-    if opts.language == "p4_14": opts.language = "p4-14"
-    if opts.language == "p4_16": opts.language = "p4-16"
+
+    ### Accept multiple ways of specifying which language,
+    ###   and ensure that it is a consistent string from now on.
+    ###
+    ### Abe`s note: before I started working on reformatting this file for
+    ###   style-guidelines compliance, the code for this part was basically:
+    ###
+    #   if  opts.language == "p4_14":
+    #       opts.language =  "p4-14"
+    #   if  opts.language == "p4_16":
+    #       opts.language =  "p4-16"
+    ###
+    ### The following should be more general, i.e. it should handle e.g.
+    ###   “p4_22” ⇒ “p4-22” just fine.
+    ###
+    ### Note: Abe dislikes the ASCII dash/minus in
+    ###   “p4-”<some decimal integer>, but is keeping it here so as to not
+    ###   break/change this part of the contract/interface between the driver
+    ###   and the rest of the compiler.  One of these days/years,
+    ###   Abe suggests switching to e.g. “P4₁₄” and “P4₁₆” instead... ;-)
```

Comment:
perhaps, but I have no idea how to type these symbols on my keyboard

Reply:
Some good pages for copy+paste...

[English Wikipedia](https://en.wikipedia.org/wiki/Unicode_subscripts_and_superscripts)
[https://unicode-table.com/en/sets/superscript-and-subscript-letters/](https://unicode-table.com/en/sets/superscript-and-subscript-letters/)
[https://unicode-table.com/en/blocks/superscripts-and-subscripts/](https://unicode-table.com/en/blocks/superscripts-and-subscripts/)
[https://www.fileformat.info/info/unicode/block/superscripts_and_subscripts/list.htm](https://www.fileformat.info/info/unicode/block/superscripts_and_subscripts/list.htm)
[https://lingojam.com/SubscriptGenerator](https://lingojam.com/SubscriptGenerator)

        

A PDF from the Unicode standard itself (also usable as a source for copy+paste): [Superscripts and Subscripts](https://unicode.org/charts/PDF/U2070.pdf)

        

... and last but perhaps not least, here is a collection of free-of-charge keymap files for entering Unicode on recent-enough versions of MS Windows; with the one of these that I have been using recently, holding down the (Windows95) menu/context key and pressing the “1!” key gives me ‘₁’, and holding down the (Windows95) menu/context key and pressing the “6^” key gives me ‘₆’...

[http://ilyaz.org/keyboard/](http://ilyaz.org/keyboard/)  (currently redirects to “https://math.berkeley.edu/~serganov/ilyaz.org/keyboard/”.)